sway/criteria.c

```diff
@@ -273,7 +273,7 @@ static int regex_cmp(const char *item, const pcre *regex) {
 
 // test a single view if it matches list of criteria tokens (all of them).
 static bool criteria_test(struct sway_container *cont, list_t *tokens) {
-	if (cont->type != C_VIEW) {
+	if (cont->type < C_CONTAINER) {
```

Comment:
Don't do numeric comparisons on enum types, enumerate the valid options and check for equality